src/scenario/outdoors.cpp

```diff
@@ -199,3 +200,21 @@ bool cOutdoors::cWandering::isNull(){
 void cOutdoors::reattach(cScenario& scen) {
 	scenario = &scen;
 }
+
+bool cOutdoors::is_city_on_map(short tile_index)
+{
+	return city_locs[tile_index].spec >= 0 && city_locs[tile_index].x < AREA_MEDIUM;
+}
+
+bool cOutdoors::is_tile_on_edge(short tile_index)
```

Comment:
I think this is better added to `cArea` and changed to take a `location` instead of an index into `city_locs`.